src/quickFixProvider.ts

```diff
@@ -1,24 +1,71 @@
 // Copyright (c) jdneo. All rights reserved.
 // Licensed under the GNU LGPLv3 license.
 
-import { CodeAction, CodeActionContext, CodeActionKind, CodeActionProvider, Range, Selection, TextDocument } from 'vscode';
+import { CodeAction, CodeActionContext, CodeActionKind, CodeActionProvider, Diagnostic, Range, Selection, TextDocument } from 'vscode';
 import { CheckstyleExtensionCommands } from './constants/commands';
 import { isQuickFixAvailable } from './utils/quickFixUtils';
 
 class QuickFixProvider implements CodeActionProvider {
     public provideCodeActions(document: TextDocument, _range: Range | Selection, context: CodeActionContext): CodeAction[] {
+        const diagnosticsByCode: IDiagnosticsByCode = groupIDiagnosticsByCode(context.diagnostics);
         const codeActions: CodeAction[] = [];
-        for (const diagnostic of context.diagnostics) {
-            if (!isQuickFixAvailable(diagnostic.code)) {
+        for (const code of Object.keys(diagnosticsByCode)) {
+            if (!isQuickFixAvailable(code)) {
                 continue;
             }
+            const diagnostics: Diagnostic[] = diagnosticsByCode[code];
+            if (diagnostics.length === 1) {
+                const diagnostic: Diagnostic = diagnostics[0];
+                codeActions.push({
+                    title: `Fix '${diagnostic.message}'`,
+                    diagnostics,
+                    command: {
+                        title: 'Fix the Checkstyle violation',
+                        command: CheckstyleExtensionCommands.FIX_CHECKSTYLE_VIOLATIONS,
+                        arguments: [document.uri, [document.offsetAt(diagnostic.range.start)], [diagnostic.code]],
+                    },
+                    kind: CodeActionKind.QuickFix,
+                });
+            } else {
+                codeActions.push({
```

Comment:
Can we leverage the implementation in the `else block` to cover the case `diagnostics.length === 1`? Seems feasible to me by introducing a utility method parsing the title according to the diagnostic length?